### PR TITLE
Machine ID: Add usage event for bot creation

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport.e/issues/1726

Depends on prehog-side PR: https://github.com/gravitational/cloud/pull/5015

### CODE REVIEW

**Comment by strideynet** [comment]
Do Not Merge until Cloud PR is merged and we can bring in the latest protos (this is why the diff drops a field from the protos).

File: proto/prehog/v1alpha/teleport.proto
Comment (strideynet) [inline proto/prehog/v1alpha/teleport.proto:272]: Looks like the cloud side protos haven't had this added and when I've copied this across from cloud, it's had this line removed, @rudream is this intentional ? do we need to add this value in the cloud repo ?

Comment (strideynet) [inline proto/prehog/v1alpha/teleport.proto:272]: Once https://github.com/gravitational/cloud/pull/4972 is merged, I can re-update this proto and this field won't be removed.

File: proto/prehog/v1alpha/teleport.proto
Comment (smallinsky) [inline proto/prehog/v1alpha/teleport.proto:667]: Are you going to extend the `ConvertUsageEvent` in another PR ?

File: proto/prehog/v1alpha/teleport.proto
Comment (smallinsky) [reply proto/prehog/v1alpha/teleport.proto:272]: I'm bit confused. This field was added in https://github.com/gravitational/teleport/pull/27518/files#diff-53145009facaad79b5bba3127f8e08dad99c90dda4434c6b927f1d4100f2a2b3R272 and was backported to v13. Why we need gravitational/cloud#4972 ?

File: proto/prehog/v1alpha/teleport.proto
Comment (strideynet) [reply proto/prehog/v1alpha/teleport.proto:667]: This event isn't submitted externally to the auth server, so I don't think `ConvertUsageEvent` needs extending ?

File: proto/prehog/v1alpha/teleport.proto
Comment (strideynet) [reply proto/prehog/v1alpha/teleport.proto:272]: I'm a little unclear as to which repo holds the source of truth from these protos, but it seems to make sense to me that this would be the cloud repo since that's where the server-side implementation is. If `teleport` is the source of truth, I can add this line back in.

Comment (smallinsky) [reply proto/prehog/v1alpha/teleport.proto:272]: > The protobuf definitions are in github.com/gravitational/cloud/build/prehog/proto/prehog and are manually synchronized with the ones in github.com/gravitational/teleport/proto/prehog

The `manually synchronized` probably means copy and paste and since the gravitational/cloud#4972 is already merged I think that we should keep the  `DISCOVER_RESOURCE_SAML_APPLICATION` def.

**Review by smallinsky — APPROVED** [review]

**Comment by strideynet** [comment]
Issues with PR in cloud not having been merged are now resolved.

File: lib/auth/bot.go
Comment (espadolini) [inline lib/auth/bot.go:176]:
```suggestion
		UserName:    authz.ClientUsername(ctx),
```

**Review by espadolini — APPROVED** [review]

**Pull request merged**
